src/uas/UAS.cc

```diff
@@ -242,6 +242,10 @@ UAS::UAS(MAVLinkProtocol* protocol, int id) : UASInterface(),
 */
 UAS::~UAS()
 {
+    stopHil();
+    // wait for the simulator to exit
+    simulation->wait();

```

Comment:
@LorenzMeier This crashes when you are not doing a simulation since simulation is NULL. Also not sure about the wait here not waiting forever. What is causing the simulation thread to close such that wait will return if I just shutdown QGC without first stopping the simulation from the widget.